tensorflow/compiler/mlir/tosa/tosa_compiler.cc

```diff
@@ -0,0 +1,69 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+// Implements the command line TOSA compiler
+#include <iostream>
+
+#include "absl/strings/str_split.h"
+#include "llvm/ADT/None.h"
+#include "llvm/ADT/StringExtras.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/FormatVariadic.h"
+#include "llvm/Support/InitLLVM.h"
+#include "llvm/Support/SourceMgr.h"
+#include "llvm/Support/ToolOutputFile.h"
+#include "mlir/Dialect/Shape/IR/Shape.h"
+#include "mlir/Dialect/Tosa/IR/TosaOps.h"
+#include "mlir/IR/AsmState.h"
+#include "mlir/IR/Diagnostics.h"
+#include "mlir/IR/Dialect.h"
+#include "mlir/IR/Function.h"
+#include "mlir/IR/MLIRContext.h"
+#include "mlir/IR/Module.h"
+#include "mlir/InitAllDialects.h"
+#include "mlir/InitAllPasses.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Support/FileUtilities.h"
+#include "mlir/Support/MlirOptMain.h"
+#include "tensorflow/compiler/mlir/init_mlir.h"
+#include "tensorflow/compiler/mlir/lite/ir/tfl_ops.h"
+#include "tensorflow/compiler/mlir/tensorflow/dialect_registration.h"
+#include "tensorflow/compiler/mlir/tosa/transforms/register_passes.h"
+#include "tensorflow/core/platform/init_main.h"
+#include "tensorflow/core/platform/logging.h"
```

Comment:
Double check that all these headers are really needed here.

Reply:
One of the open questions here is do we need tosa_compiler.cc and the binary at all ? We kept it around until there was clarity on integrating these passes into tf-opt . If the use of tf-opt is ok, then we don't need a separate pass manager tosa-opt , and we can eliminate this altogether.

Reply:
This is a good question.
It is convenient to have `tf-tfjs-opt` or `tf-tfrt-opt` because when iterating on development and testing, it is faster to build/relink a smaller subset of the project. For example when I test `tensorflow/compiler/mlir/mhlo/...` we won't build all of TF (actually almost none of it).

In this case it may not be worth a separate tool, except that for now I believe @stellaraccident is trying to keep this separated from our main CI until the integration is complete and "stable" (just a few weeks I guess).
If we integrate it in `tf-opt`, then I'd suggest using a build option `bazel build --with-tosa=true` or something like that, which default to off for now.

Reply:
Sure, I'll attempt to merge in the WIP conditional build pieces and try this out.